File: exercises/08_ml/r/01_regression.R
# 対訳 — Must 01 — 線形回帰（Culmen + Flipper → Body_Mass）
# Culmen_LengthとFlipper_LengthからBody_Massを予測し、テストデータのMSEとMAPEを表示してください。
library(readr)
library(dplyr)

path <- "../../../data/penguins.csv"
df <- read_csv(path, show_col_types = FALSE) |>
  tidyr::drop_na(Culmen_Length, Flipper_Length, Body_Mass)

set.seed(123)
n <- nrow(df)
test_idx <- sample.int(n, size = floor(0.3 * n))
train <- df[-test_idx, ]
test <- df[test_idx, ]

fit <- lm(Body_Mass ~ Culmen_Length + Flipper_Length, data = train)
y_hat <- predict(fit, newdata = test)

mse <- mean((test$Body_Mass - y_hat)^2)
mape <- mean(abs((test$Body_Mass - y_hat) / test$Body_Mass))
cat("MSE:", round(mse, 3), "\n")
cat("MAPE:", round(mape, 4), "\n")

File: exercises/08_ml/python/01_regression.py
"""【問題】`Culmen_Length`と`Flipper_Length`から`Body_Mass`を
`LinearRegression()`で予測し、テストデータのMSEとMAPEを表示してください。
"""
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_percentage_error, mean_squared_error
from sklearn.model_selection import train_test_split

df = pd.read_parquet("../../../data/penguins.parquet").dropna(
    subset=["Culmen_Length", "Flipper_Length", "Body_Mass"]  # 欠損を除去
)

# TODO: 特徴量Xと目的変数yを用意
# TODO: train_test_split（test_size=0.3, random_state=123）
# TODO: LinearRegressionで学習・予測し、MSEとMAPEを表示
